centreon-certified/prometheus/prometheus-pushgateway-events-apiv2.lua

```diff
@@ -0,0 +1,378 @@
+    state_type          = event.state_type,
+    hostname            = event.cache.host.name,
+    service_description = sdesc,
+    output              = event.output,
+  }
+end
+
+--------------------------------------------------------------------------------
+-- EventQueue:add, add an event to the sending queue
+--------------------------------------------------------------------------------
+function EventQueue:add()
+  -- store event in self.events lists
+  local category = self.sc_event.event.category
+  local element = self.sc_event.event.element
+
+  self.sc_logger:debug("[EventQueue:add]: add event in queue category: " .. tostring(self.sc_params.params.reverse_category_mapping[category])
+    .. " element: " .. tostring(self.sc_params.params.reverse_element_mapping[category][element]))
+
+  self.sc_logger:debug("[EventQueue:add]: queue size before adding event: " .. tostring(#self.sc_flush.queues[category][element].events))
+  self.sc_flush.queues[category][element].events[#self.sc_flush.queues[category][element].events + 1] = self.sc_event.event.formated_event
+
+  self.sc_logger:info("[EventQueue:add]: queue size is now: " .. tostring(#self.sc_flush.queues[category][element].events) 
+    .. ", max is: " .. tostring(self.sc_params.params.max_buffer_size))
+end
+
+--------------------------------------------------------------------------------
+-- EventQueue:build_payload, concatenate data so it is ready to be sent
+-- @param payload {string} json encoded string
+-- @param event {table} the event that is going to be added to the payload
+-- @return payload {string} json encoded string
+--------------------------------------------------------------------------------
+function EventQueue:build_payload(payload, event)
+  if not payload then
+    payload = event --TBD
+  else
+    self.sc_logger:error("[EventQueue:build_payload]: payload should be nil at this point.")
+    table.insert(payload, event) --TBD
+  end
+  
+  return payload
+end
+
+
+function EventQueue:send_data(payload, queue_metadata)
+  self.sc_logger:debug("[EventQueue:send_data]: Starting to send data")
+  --self.sc_logger:warning("[EventQueue:send_data]: payload: " .. self.sc_common:dumper(payload))
+  local data = ""
+  local httpResponseBody = ""
+  local label = "status"
+
+  local name = convert_to_openmetric(payload.prom_hname .. '_' .. payload.prom_sdesc .. ':' .. label .. ':monitoring_status')
+  data = data .. '# TYPE ' .. name .. ' counter\n'
+  data = data .. '# HELP ' .. name .. ' 0 is OK, 1 is WARNING, 2 is CRITICAL, 3 is UNKNOWN\n'
+  if not payload.hostgroupsLabel then
+    data = data .. name .. '{label="monitoring_status", host="' .. payload.prom_hname .. '", service="' .. payload.prom_sdesc .. '"} ' .. payload.state .. '\n'
+  else
+    data = data .. name .. '{label="monitoring_status", host="' .. payload.prom_hname .. '", service="' .. payload.prom_sdesc .. '", ' ..  payload.hostgroupsLabel .. '} ' .. payload.state .. '\n'
+  end
+
+
```

Comment:
missing the send_data_test and log_curl_command features

Reply:
fixed here https://github.com/centreon/centreon-stream-connector-scripts/pull/279/commits/b386ac65b9d1c10786fe7a9be34ca8fb570f334e